src/printer.js

```diff
@@ -1026,7 +1026,165 @@ function printLines(path, options, print, childrenAttribute = "children") {
+  // ex: $test = new class($arg1, $arg2) extends TestClass {};
+  if (
+    node.isAnonymous &&
+    parentNode.kind === "new" &&
+    parentNode.arguments.length > 0
+  ) {
+    declaration.push(printArgumentsList(path, options, print));
+  }
+
+  const partsDeclarationGroup = [];
+
+  if (node.extends) {
+    if (!Array.isArray(node.extends)) {
+      partsDeclarationGroup.push(
+        group(
+          concat([
+            // check if the extends node has a comment
+            hasDanglingComments(node.extends)
+              ? concat([
+                  hardline,
+                  path.call(
+                    extendsPath =>
+                      comments.printDanglingComments(
+                        extendsPath,
+                        options,
+                        true
+                      ),
+                    "extends"
+                  ),
+                  hardline
+                ])
+              : line,
+            "extends ",
+            path.call(print, "extends")
+          ])
+        )
+      );
+    } else {
+      partsDeclarationGroup.push(
+        concat([
+          line,
+          "extends",
+          group(
+            indent(
+              concat([
+                join(
+                  ",",
+                  path.map(extendPath => {
+                    // check if any of the implements nodes have comments
+                    return hasDanglingComments(extendPath.getValue())
+                      ? concat([
+                          hardline,
+                          comments.printDanglingComments(
+                            extendPath,
+                            options,
+                            true
+                          ),
+                          hardline,
+                          print(extendPath)
+                        ])
+                      : concat([line, print(extendPath)]);
+                  }, "extends")
+                )
+              ])
+            )
+          )
+        ])
+      );
+    }
+  }
+
+  if (node.implements) {
+    partsDeclarationGroup.push(
+      concat([
+        node.kind === "class" && node.isAnonymous ? " " : line,
+        "implements",
+        group(
+          indent(
+            concat([
+              join(
+                ",",
+                path.map(implementPath => {
+                  // check if any of the implements nodes have comments
+                  return hasDanglingComments(implementPath.getValue())
+                    ? concat([
+                        hardline,
+                        comments.printDanglingComments(
+                          implementPath,
+                          options,
+                          true
+                        ),
+                        hardline,
+                        print(implementPath)
+                      ])
+                    : concat([line, print(implementPath)]);
+                }, "implements")
```

Comment:
Now we respect comments in implements and extends for `class/interface/trait`, tests will add in other PR